src/main/java/org/example/builder/AnimalBuilder.java

```diff
@@ -0,0 +1,82 @@
+package org.example.builder;
+
+import org.example.animal.AnimalEntity;
+import org.example.animal.AnimalType;
+import org.example.components.*;
+
+import java.util.Objects;
+
+public class AnimalBuilder implements Builder {
+    private AnimalType animalType;
+    private String origin;
+    private Body body;
+    private Fur fur;
+    private Size size;
+    private Tail tail;
+
+    @Override
+    public void setAnimalType(final AnimalType animalType) {
+        this.animalType = animalType;
+    }
+
+    @Override
+    public void setBody(final Body body) {
+        this.body = Objects.isNull(body) ? ComponentsStorage.getInstance().getDefaultBody() : body;
```

Comment:
For me it's more implicite
```suggestion
        Objects.requireNotNull(body)
        this.body = body;
```